Cleanups for starting on FD

nilsdeppe opened the pull request:
## Proposed changes

Various cleanups for moving mesh and work towards starting on the FD grid instead of the DG grid.

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsdeppe committed:
Fix compilation error in InterpolateFromSpec

nilsdeppe committed:
Improve error message in NeighborRdmpAndVolumeData

nilsdeppe committed:
Remove static mesh ASSERT from GhValenciaDivClean

nilsdeppe committed:
Move DG-FD static mesh asserts into systems

nilsdeppe committed:
Clean up Jacobians.hpp

- Remove incorrect comments
- Remove unused include
- Add missing include
- Value to reference arg

nilsdeppe committed:
Add DetInvJac from logical to inertial for FD

nilsdeppe committed:
Clean up jacobians in FD solver

nilsdeppe committed:
Support DG in SetInitialRdmpData

nilsdeppe committed:
Add logical to inertial jacobians to FD initialization

nilsdeppe commented:
I'm ignoring the clang-tidy failure because in the followup PR I will be deleting that code anyway.

nilsvu commented on src/PointwiseFunctions/AnalyticData/GeneralRelativity/InterpolateFromSpec.hpp:
```diff
@@ -82,18 +82,19 @@ tuples::tagged_tuple_from_typelist<Tags> interpolate_from_spec(
       const auto component_name =
           tensor.component_name(tensor.get_tensor_index(component_i));
-      if constexpr (tensor.rank() == 1) {
+      if constexpr (std::decay_t<decltype(tensor)>::rank() == 1) {
```
This didn't show up on Wheeler or CaltechHPC with GCC. Does clang not support this or something like that?

nilsdeppe replied:
I don't know what the standard says, but Clang tends to be right more often than GCC when it comes to standard C++.

nilsvu commented on src/Evolution/Systems/GrMhd/GhValenciaDivClean/Subcell/NeighborPackagedData.hpp:
```diff
@@ -88,12 +88,6 @@ struct NeighborPackagedData {
-           "Haven't yet added support for moving mesh to DG-subcell. This "
-           "should be easy to generalize, but we will want to consider "
-           "storing the mesh velocity on the faces instead of "
-           "re-slicing/projecting.");
```
Add code comment with this possible optimization?

nilsdeppe replied:
I'll add it to the dox since I worry that the code comment will get lost :)

nilsvu commented on src/Evolution/DgSubcell/Tags/Jacobians.hpp:
```diff
@@ -39,6 +39,13 @@ struct DetInverseJacobianLogicalToGrid : db::SimpleTag {
 
+/// \brief The determinant of the inverse Jacobian from the element logical
+/// frame to the inertial frame at the cell centers.
+struct DetInverseJacobianLogicalToInertial : db::SimpleTag {
```
Want to use this in `Evolution/DgSubcell/Tags/ObserverCoordinates.hpp` (in `ObserverJacobianAndDetInvJacobian`)? (Side note: those Jacobian compute tags probably need a `Compute` suffix and should move to their own file)

nilsdeppe replied:
Those tags also support `Grid->Inertial` maps, which isn't yet generalized. Is it okay if I leave that for now?

nilsvu replied:
Sure

nilsvu commented:
Squash directly if you do these

nilsdeppe committed:
Add Compute suffix to ObserverJacobianAndDetInvJacobian

nilsdeppe commented:
I rebased, squashed the easy things, and added a commit that renames `ObserverJacobianAndDetInvJacobian` to `ObserverJacobianAndDetInvJacobianCompute`.

Thanks for the reviews!

nilsvu commented:
Unrelated failure in translation map
